vault/data_source_generic_secret.go

```diff
@@ -22,6 +22,13 @@ func genericSecretDataSource() *schema.Resource {
 				Description: "Full path from which a secret will be read.",
 			},
 
+			"version": {
+				Type:     schema.TypeInt,
+				Required: false,
+				Optional: true,
+				Default:  -1,
```

Comment:
This is actually to retrieve the latest version of the secret from Vault, rather than the Terraform state version. I've extracted it to a constant to (hopefully) better represent what it does.